src/blockdiagonal.jl

```diff
@@ -22,15 +25,15 @@ BlockDiagonal(B::BlockDiagonal) = B
 is_square(A::AbstractMatrix) = size(A, 1) == size(A, 2)
 
 """
-    blocks(B::BlockDiagonal{T, V}) -> Vector{V}
+    blocks(B::BlockDiagonal{T, V}) -> V
 
 Return the on-diagonal blocks of B.
 """
 blocks(B::BlockDiagonal) = B.blocks
 
 # BlockArrays-like functions
 """
-    blocksizes(B::BlockDiagonal) -> Vector{Tuple}
+    blocksizes(B::BlockDiagonal{T, V}) -> V
```

Comment:
Do we want this to return a Vector or a Tuple depending on the block type? Or should we force it to return a vector/tuple always?

Reply:
my thinking is that we'd use `blocks::Tuple` for performance reasons when you've a small number of heterogenous `blocks`, in which case getting small Tuples from methods that operate on blocks will also be slightly better for performance, so no reason to force a `Vector` to be returned. It doesn't make sense to force a Tuple return in general, since there could be many elements.